src/app/signup/member/page.tsx

```diff
@@ -0,0 +1,388 @@
+"use client";
+import { useState } from "react";
+import { Select, Input, Typography, Button, Space, Form, Radio, Flex, Row, Col, Checkbox } from "antd";
+import { useForm, Controller, FormProvider, useFormContext } from "react-hook-form";
+import { yupResolver } from "@hookform/resolvers/yup";
+import { UserFormKey } from "@/constant/common/formKey";
+import { signUpFormSchema } from "@/schema/signup/signUpFormSchema";
+import { district, city, exerciseList, fieldList, type OptionValue } from "@/constant/common/signup";
+
+interface IMemberFormData {
+  [UserFormKey.NICKNAME]: string;
+  [UserFormKey.CELLPHONE]: string;
+  [UserFormKey.SEX]: string;
+  [UserFormKey.AGE]: string;
+  [UserFormKey.CITY]: string;
+  [UserFormKey.DISTRICT]: string;
+  [UserFormKey.GOALS]: string[];
+  [UserFormKey.CATEGORY]: string[];
+}
+
+interface StepProps {
+  next?: () => void;
+  prev?: () => void;
+}
+const { Title, Text } = Typography;
+
+const Step1 = ({ next }: StepProps) => {
+  const {
+    control,
+    formState: { errors, isValid },
+    watch,
+  } = useFormContext<IMemberFormData>();
+  const watchedName = watch(UserFormKey.NICKNAME);
+  const watchedPhone = watch(UserFormKey.CELLPHONE);
+
+  return (
+    <>
+      <Controller
+        name={UserFormKey.NICKNAME}
+        control={control}
+        render={({ field, fieldState }) => (
+          <>
+            <Form.Item
+              style={{ width: "100%" }}
+              validateStatus={errors[UserFormKey.NICKNAME] ? "error" : ""}
+              help={errors[UserFormKey.NICKNAME] ? errors[UserFormKey.NICKNAME]?.message : ""}
+            >
+              <Typography>닉네임</Typography>
+              <Input
+                {...field}
+                placeholder="특수 문자 제외(2~5글자)"
+                id="nickname-input"
+                value={field.value}
+                onChange={field.onChange}
+                status={fieldState.error && "error"}
+              />
```

Comment:
```suggestion
              <Typography>닉네임</Typography>
              <Input
                {...field}
                placeholder="특수 문자 제외(2~5글자)"
                id="nickname-input"
                status={fieldState.error && "error"}
              />
```

위에 field 관련해서 spread 문법을 사용하고 있는데 value랑 onChange 또 따로 선언해준 이유가 있을까용?

Reply:
이전 컴포넌트 (MUI) 에서 사용하던 패턴이 남아있었습니다...! 스프레드 문법 제거 후 필요한 props  사용으로 수정했습니다!